packages/relay/src/lib/eth.ts

```diff
@@ -1932,16 +1932,29 @@ export class EthImpl implements Eth {
   }
 
   private async getAccountLatestEthereumNonce(address: string, requestId?: string) {
-    // check if address is a valid contract then get ethereumNonce from consensus node until HIP 729 is implemented
-    const validContract = await this.mirrorNodeClient.isValidContract(address, requestId);
-    if (validContract)  {
+    const cachedLabel = `${constants.CACHE_KEY.RESOLVE_ENTITY_TYPE}_${address}`;
```

Comment:
Isn't this approach essentially extracting the logic out?
`this.mirrorNodeClient.isValidContract` has caching logic.

In which scenario do we reduce query attempts to the Mirror Node?
`constants.CACHE_KEY.RESOLVE_ENTITY_TYPE` currently caches for 1hr.
`isValidContract` is caching for 1 day and was supposed to be an improvement.